support/get-logs.sh

```diff
@@ -10,54 +10,123 @@
     printf "  -r | -rel | -release     specify the required RELEASE\n\n"
+    printf "  -s | -since              specify how many hours back the logs should go (default 120)n"
     printf "\nPre-conditions:\n"
     printf "  1) kubectl/oc must be installed\n"
     printf "  2) (preferred) helm & openssl installed\n"
     printf "  3) User must be logged into host as cluster-admin\n"
     printf "  4) User must know the namespace that Event Streams is deployed in, and the name of that release\n\n"
 }
 
-declare -a ES_COMPONENTS=(
-    "security"
-    "collector"
-    "elastic"
-    "indexmgr"
-    "kafka"
-    "proxy"
-    "rest"
-    "restproducer"
-    "restproxy"
-    "replicator"
-    "schemaregistry"
-    "ui"
-    "zookeeper"
-    "essential"
+# prints the incoming to the terminal and writes it to the output.log
+printAndLog () {
+    tee -a "${LOGDIR}/output.log"
+}
+
+# only print valid printable characters
+cleanOutput () {
+    tr -cd '\11\12\15\40-\176'
+}
+
+# print the supplied text in red by setting the forground colour to red then unsetting it
+printRed () {
+    tput setaf 1
+    printf "${@}"
+    tput sgr0
+}
+
+# print the supplied text in green
+printGreen () {
+    tput setaf 2
+    printf "${@}"
+    tput sgr0
+}
+
+# print the supplied text in yellow
+printYellow () {
+    tput setaf 3
+    printf "${@}"
+    tput sgr0
+}
+
+# print the supplied text in red and write it to the log, note `printRed blah | printAndLog` will give formatting problems in the log
+printRedAndLog () {
+    tput setaf 1
+    printf "${@}" | printAndLog
+    tput sgr0
+}
+
+# print the supplied text in green and write it to the log
+printGreenAndLog () {
+    tput setaf 2
+    printf "${@}" | printAndLog
+    tput sgr0
+}
+
+# print the supplied text in yellow and write it to the log
+printYellowAndLog () {
+    tput setaf 3
+    printf "${@}" | printAndLog
+    tput sgr0
+}
+
+# print the standard green done message
+printDone () {
+    printGreen "\t[DONE]\n"
+}
+
+# print the standard green done message and write it to the log
+printDoneAndLog () {
+    printGreenAndLog "\t[DONE]\n"
+}
+
+####################################################################################################
+# Configuration
+####################################################################################################
+
+declare -a ES_HELM_COMPONENT_LABELS=(
+    "component=collector"
+    "component=elastic"
+    "component=essential"
+    "component=indexmgr"
+    "component=kafka"
+    "component=proxy"
+    "component=replicator"
+    "component=rest"
+    "component=restproducer"
+    "component=restproxy"
+    "component=schemaregistry"
+    "component=security"
+    "component=ui"
+    "component=zookeeper"
 )
 
 declare -a RESOURCES=(
-    "nodes"
     "networkpolicies"
     "services"
-    "persistentvolumes"
     "persistentvolumeclaims"
     "configmaps"
     "statefulsets"
     "deployments"
     "secrets"
-    "storageclasses"
```

Comment:
What's the rational for removing?

Reply:
because it got moved into global resources

The old logic had global resources in the resources and then used the global resources as a sub check of resources

I thought this was unnecessarily confusing so now we just have 2 lists global and not global and the two are independent

Reply:
OK fair enough